### ISSUE TITLE
Improvements to Posts Block

### ISSUE THREAD

**Issue by JohnPixle** [Codeinwp/otter-blocks#655]
@HardeepAsrani 👋🏻  🙃
I finally had the chance to consolidate my ideas on improving the Posts block. I have created the [Figma mockups](https://www.figma.com/file/LRrhO1ovZnXFibXVsezR8a/?node-id=318%3A2) and thought it would be more efficient to also create a quick walkthrough to elaborate on the improvements. Feel free to leave your comments in FIgma or here (or both), whatever works best for you, and we can take it from there, based on what you think it is possible to improve at this point, and what we can postpone for a future version

https://www.loom.com/share/701b8a9e4e614161b7ae13761464dfa3

**Comment by HardeepAsrani** [comment]
@JohnPixle Thanks for the video. I saw it and seem to like all the idea.

But I'm a bit confused about the entire Posts Block. Have you tried the new Core's Query Block that comes with WordPress 5.8? It's a faaaar more advance version of our block and I wonder if we should switch to that type of a block (where each element is a block of its own) or just continue to make Posts Block as a more advance version of a simple Posts Block rather than splitting everything to their own block?

**Comment by HardeepAsrani** [comment]
Initial idea was to make Posts Block like Core's Query Block but I don't know how much better we can make it. It will be mostly the same or a more complex version doing mostly the same. Don't see us adding much value there as Query Block is pretty advanced and using all the goods that Core has to offer.

If we were extending that to allow more options that would've been a different thing but don't think we can beat the Query Block.

Here's a small video showing the Query: https://www.loom.com/share/ab9ac86398e644a0b356f3be6250402c

**Comment by JohnPixle** [comment]
Thanks for the insight here @HardeepAsrani !
Indeed I have seen the core's Query block, and it is very flexible. However there are crucial styling issues with Neve at this point, that need to be tackled. I had created an issue about that [here.](https://github.com/Codeinwp/neve-pro-addon/issues/1722)

There are a couple of things in my mind. 

Firstly, I feel we should allow some time for the Core query block to mature, and see if they bring any more changes to the table before attempting to approach it and fix any issues with the theme. With the FSE overall, I feel the changes are too many and fast-paced for us to catch up with, so I tend to have a more conservative approach on this.

Regarding the Posts block in Otter, my initial idea was to improve the visual aspect of it primarily, and add some of those options I highlighted in the video. So the overall idea would be to offer the users an improved experience of using the block, and give them these pre-configured layout options instead of splitting everything to it's own block.

Perhaps what we could do, is to continue improving the post block as it is, and perhaps at some time in the future we could explore the potential of adding the nested-block-based solutions in Otter, but independently from the Posts block.

(kinda like the Core uses the Latest posts block, as well as the individual post block elements)

Let me know your thoughts!

**Comment by HardeepAsrani** [comment]
@JohnPixle I agree with this, let's see how we can much we can improve it without turning it into an over-complicated block.

**Comment by Soare-Robert-Daniel** [comment]
@JohnPixle I have some questions about the design:

I implemented the display for categories as inline with the other metadata: 
![image](https://user-images.githubusercontent.com/17597852/150537491-14cd3509-3c33-4c54-8486-5f4e6de98aa5.png)

But what if there are more categories? should I display only the first?

Also, how does the displaying of the comments look (if we are going to implement this)?

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel Yea, we can just show one category. Can there be a way to allow users to put some kind of pattern in meta field so they can decide how to display them? Like: `Posted in %cat% by %author% on %date%` something like this? What do you think @JohnPixle?

**Comment by JohnPixle** [comment]
@HardeepAsrani In this case, we can/should remove/edit the entire panel for the post meta controls, correct?
![meta](https://user-images.githubusercontent.com/4193672/150998310-95c16952-e322-4b92-9251-e1a8cb1186ab.png)

Having both of these UX patterns would cause some struggle as I envision it. 
I do agree though that we need some kind of further flexibility on the order of the post meta as well (something that your suggestion solves for). 

Regarding the categories, I am also ok with showing only one category (primary).

**Comment by HardeepAsrani** [comment]
Yea, instead it can just be a text field with maybe info how to use it and a sane default for the current format.
